nucypher/config/storages.py

```diff
@@ -0,0 +1,307 @@
+        payload = {self._TYPE_LABEL: self._name}
+        return payload
+
+    @classmethod
+    def from_payload(cls, payload: str, *args, **kwargs) -> 'InMemoryNodeStorage':
+        if payload[cls._TYPE_LABEL] != cls._name:
+            raise cls.NodeStorageError
+        return cls(*args, **kwargs)
+
+    def initialize(self) -> None:
+        self.__known_nodes = dict()
+
+
+class LocalFileBasedNodeStorage(NodeStorage):
+
+    _name = 'local'
+    __FILENAME_TEMPLATE = '{}.node'
+    __DEFAULT_DIR = os.path.join(DEFAULT_CONFIG_ROOT, 'known_nodes', 'metadata')
+
+    class NoNodeMetadataFileFound(FileNotFoundError, NodeStorage.UnknownNode):
+        pass
+
+    def __init__(self,
+                 known_metadata_dir: str = __DEFAULT_DIR,
+                 *args, **kwargs
+                 ) -> None:
+
+        super().__init__(*args, **kwargs)
+        self.log = getLogger(self.__class__.__name__)
+        self.known_metadata_dir = known_metadata_dir
+
+    def __generate_filepath(self, checksum_address: str) -> str:
+        metadata_path = os.path.join(self.known_metadata_dir, self.__FILENAME_TEMPLATE.format(checksum_address))
+        return metadata_path
+
+    def __read(self, filepath: str, federated_only: bool):
+        from nucypher.characters.lawful import Ursula
+        try:
+            with open(filepath, "rb") as seed_file:
+                seed_file.seek(0)
+                node_bytes = self.deserializer(seed_file.read())
+                node = Ursula.from_bytes(node_bytes, federated_only=federated_only)
+        except FileNotFoundError:
+            raise self.UnknownNode
+        return node
+
+    def __write(self, filepath: str, node):
+        with open(filepath, "wb") as f:
+            f.write(self.serializer(self.character_class.__bytes__(node)))
+        self.log.info("Wrote new node metadata to filesystem {}".format(filepath))
+        return filepath
+
+    def all(self, federated_only: bool) -> set:
+        filenames = os.listdir(self.known_metadata_dir)
+        self.log.info("Found {} known node metadata files at {}".format(len(filenames), self.known_metadata_dir))
+        known_nodes = set()
+        for filename in filenames:
+            metadata_path = os.path.join(self.known_metadata_dir, filename)
+            node = self.__read(filepath=metadata_path, federated_only=federated_only)   # TODO: 466
+            known_nodes.add(node)
+        return known_nodes
+
+    def get(self, checksum_address: str, federated_only: bool):
+        metadata_path = self.__generate_filepath(checksum_address=checksum_address)
+        node = self.__read(filepath=metadata_path, federated_only=federated_only)   # TODO: 466
+        return node
+
+    def save(self, node):
+        try:
+            filepath = self.__generate_filepath(checksum_address=node.checksum_public_address)
+        except AttributeError:
+            raise AttributeError("{} does not have a rest_interface attached".format(self))  # TODO.. eh?
```

Comment:
Wanna clarify here?

Reply:
There are times when we have a character without a checksum address associated, although, this seems out of place, and was derived from the original file-only storages.  Id like to remove it, but will need some tests to show that this condition is not possible.